test/common/upstream/load_balancer_fuzz.cc

```diff
@@ -0,0 +1,123 @@
+#include "test/common/upstream/load_balancer_fuzz.h"
+
+#include "test/common/upstream/utility.h"
+
+namespace Envoy {
+namespace Upstream {
+
+// Anonymous namespace for helper functions
+namespace {
+std::vector<uint64_t>
+constructByteVectorForRandom(test::common::upstream::LoadBalancerTestCase input) {
+  std::vector<uint64_t> byteVector;
+  for (int i = 0; i < input.bytestring_for_random_calls().size(); ++i) {
+    byteVector.push_back(input.bytestring_for_random_calls(i));
+  }
+  return byteVector;
+}
+} // namespace
+
+// Required because super class constructor has logic
+LoadBalancerFuzzBase::LoadBalancerFuzzBase() : LoadBalancerFuzzTestBase() {}
+
+void LoadBalancerFuzzBase::initializeFixedHostSets(uint32_t num_hosts_in_priority_set,
+                                                   uint32_t num_hosts_in_failover_set) {
+  // TODO: Cap on ports?
+  int port = 80;
+  for (uint32_t i = 0; i < num_hosts_in_priority_set; ++i) {
+    host_set_.hosts_.push_back(makeTestHost(info_, "tcp://127.0.0.1:" + std::to_string(port)));
+    ++port;
+  }
+  for (uint32_t i = 0; i < num_hosts_in_failover_set; ++i) {
+    failover_host_set_.hosts_.push_back(
+        makeTestHost(info_, "tcp://127.0.0.1:" + std::to_string(port)));
+    ++port;
+  }
+  // TODO: More than two hosts?
+}
+
+void LoadBalancerFuzzBase::initializeAndReplay(test::common::upstream::LoadBalancerTestCase input) {
+  // TODO: Keep this random instantiation even in load balancers that don't call into it, or do they
+  // all?
+  random_.bytestring_ = constructByteVectorForRandom(input);
```

Comment:
if only one field of the input matters to this function, why not scope the params to it?
(that way you can treat the function as a blackbox. if the input changes, then you don't need to go into `constructByteVectorForRandom` to update how the function accesses what it needs from `input`, you just update the parameter)

Reply:
Great point. Thanks for catching this. Does this map to action streams as well?

Reply:
Nevermind, just saw your comment on action streams!

Reply:
Changed to repeated ptr.